consensus/src/chained_bft/event_processor.rs

```diff
@@ -609,10 +609,20 @@ impl<T: Payload> EventProcessor<T> {
 
     /// Generate sync info that can be attached to an outgoing message
     fn gen_sync_info(&self) -> SyncInfo {
+        let hqc = self.block_store.highest_quorum_cert().as_ref().clone();
+        // No need to include HTC if it's lower than HQC
+        let hqc_round = hqc.certified_block_round();
+        let htc = self.pacemaker.highest_timeout_certificate().and_then(|tc| {
+            if tc.round() > hqc_round {
+                Some(tc)
+            } else {
+                None
+            }
+        });
```

Comment:
SyncInfo has 2 goals:
1) Advancing a validator to the most recent round.  Given that either a QC/TC are required but never both, we can send the (least expensive) one that justifies the change.
2) Advancing a validator to the latest commit (highest ledger info) to reduce commit latency.

What do you think about this logic should be encapsulated within the TimeoutCertificate itself however, rather than externally, so that SyncInfo always maintains this invariant?

Reply:
It's a good point. I'd prefer to not touch pacemaker / the current timeout certificates if possible because they're going to be out after #1302 anyway (fun fact, we wouldn't have this problem of quadratic TCs in the new world because the new TimeoutCertificate is a round with its signatures only).
@aching do you mind postponing the larger refactoring of the SyncInfo generation until after we're done with the timeouts migration?